TypeTheory does not compile with Coq 8.10

File "./TypeTheory/Initiality/Typing.v", line 63, characters 2-275:
Error:
Anomaly
"Ill-formed template inductive declaration: not polymorphic on any universe."
Please report at http://coq.inria.fr/bugs/.

---

COQC TypeTheory/ALV1/CwF_Defs_Equiv.v
does not compile either because use fails somewhere

Update: this problem persists with Coq 8.10.1

```
COQC TypeTheory/ALV1/CwF_Defs_Equiv.v
File "./UniMath/Foundations/Init.v", line 180, characters 17-24:
Error: Illegal application (Non-functional construction):
The expression
 "weqtotaltoforall (λ A : Ty ?pp ?Γ : hSet, ∑ y, ?P y)
    (λ (A : Ty ?pp ?Γ : hSet)
     (ΓAπt : (λ A0 : Ty ?pp ?Γ : hSet, ∑ y, ?P@{A:=A0} y) A),
     rep1_fiber_axioms A (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))
    ?t ?x" of type
 "∑ y : (λ A : Ty ?pp ?Γ : hSet, ∑ y, ?P y) ?x,
  (λ (A : Ty ?pp ?Γ : hSet)
   (ΓAπt : (λ A0 : Ty ?pp ?Γ : hSet, ∑ y, ?P@{A:=A0} y) A),
   rep1_fiber_axioms A (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))
    ?x y" cannot be applied to the term
 "?y" : "?T"

make[1]: *** [TypeTheory/ALV1/CwF_Defs_Equiv.vo] Error 1
make: *** [all] Error 2
```

Here is the precise error message for the second problem mentioned by @rmatthes :
```
COQC TypeTheory/ALV1/CwF_Defs_Equiv.v
File "./UniMath/Foundations/Init.v", line 180, characters 17-24:
Error: Illegal application (Non-functional construction):
The expression
 "weqtotaltoforall (λ A : Ty ?pp ?Γ : hSet, ∑ y, ?P y)
    (λ (A : Ty ?pp ?Γ : hSet)
     (ΓAπt : (λ A0 : Ty ?pp ?Γ : hSet, ∑ y, ?P@{A:=A0} y) A),
     rep1_fiber_axioms A (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))
    ?t ?x" of type
 "∑ y : (λ A : Ty ?pp ?Γ : hSet, ∑ y, ?P y) ?x,
  (λ (A : Ty ?pp ?Γ : hSet)
   (ΓAπt : (λ A0 : Ty ?pp ?Γ : hSet, ∑ y, ?P@{A:=A0} y) A),
   rep1_fiber_axioms A (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))
    ?x y" cannot be applied to the term
 "?y" : "?T"

make[1]: *** [TypeTheory/ALV1/CwF_Defs_Equiv.vo] Error 1
make: *** [all] Error 2
```
I have tried to fix it, but unsuccessfully so far.

With Coq 8.10.2 it's no longer an anomaly:
```
"coqc"   -q -noinit -indices-matter -type-in-type -w none  -Q TypeTheory TypeTheory  TypeTheory/Initiality/Typing.v 
File "./TypeTheory/Initiality/Typing.v", line 63, characters 2-275:
Error: Ill-formed template inductive declaration: not polymorphic on any
universe.

make[1]: *** [build/CoqMakefile.make:659: TypeTheory/Initiality/Typing.vo] Error 1
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,7 @@
+/.coq_makefile_output.conf
+/.coqdeps.d
 *.backup
-.coq_makefile_input
+/.coq_makefile_input
 build/CoqMakefile.make
 
 ## Core latex/pdflatex auxiliary files:

--- a/.travis.yml
+++ b/.travis.yml
@@ -13,12 +13,14 @@ install:
   - eval $(opam config env)
   - opam install lablgtk camlp5 num --yes --verbose
 before_script:
-  - WPATH=$PWD
-  - cd ..        
-  - git clone https://github.com/UniMath/UniMath
-  - cd UniMath && git show && make PACKAGES="Foundations MoreFoundations Combinatorics Algebra NumberSystems CategoryTheory Topology Ktheory PAdics" install
+  - pushd ..        
+  - git clone --no-checkout https://github.com/UniMath/UniMath
+  - cd UniMath
+  - git checkout 2dadfb61f5ef0d9805cf0eb6b80ef2beb26472d5
+  - git show
+  - make PACKAGES="Foundations MoreFoundations Combinatorics Algebra NumberSystems CategoryTheory Topology Ktheory PAdics" install
   - export PATH=$PATH:$PWD/sub/coq/bin/
-  - cd $WPATH
+  - popd
 script:
   - make 
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,11 +16,10 @@ before_script:
   - pushd ..        
   - git clone --no-checkout https://github.com/UniMath/UniMath
   - cd UniMath
-  - git checkout 2dadfb61f5ef0d9805cf0eb6b80ef2beb26472d5
+  - git checkout master
   - git show
   - make PACKAGES="Foundations MoreFoundations Combinatorics Algebra NumberSystems CategoryTheory Topology Ktheory PAdics" install
   - export PATH=$PATH:$PWD/sub/coq/bin/
   - popd
 script:
   - make 
-

--- a/Makefile
+++ b/Makefile
@@ -145,7 +145,7 @@ clean::
 	find TypeTheory \( -name .\*.aux -o -name \*.glob -o -name \*.v.d -o -name \*.vo \) -delete
 	find TypeTheory -type d -empty -delete
 clean::; rm -rf $(ENHANCEDDOCTARGET)
-latex-clean clean::; cd $(LATEXDIR) ; rm -f *.pdf *.tex *.log *.aux *.out *.blg *.bbl
+latex-clean clean::; rm -rf $(LATEXDIR)
 
 distclean:: clean
 distclean::          ; - $(MAKE) -C sub/coq distclean

--- a/TypeTheory/ALV1/CwF_Defs_Equiv.v
+++ b/TypeTheory/ALV1/CwF_Defs_Equiv.v
@@ -199,8 +199,10 @@ Proof.
       (fun Γ Y => forall A, rep1_fiber_axioms A (pr2 (pr1 (Y A))) (pr2 (Y A)))).
   apply weqonsecfibers; intro Γ.
   eapply weqcomp.
-    use (@weqtotaltoforall _ _
-      (fun A ΓAπt => rep1_fiber_axioms A (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))).
+  { exact (weqtotaltoforall
+           _
+           (fun A ΓAπt => @rep1_fiber_axioms pp Γ _ (pr1 (pr1 ΓAπt)) 
+                                             (pr2 (pr1 ΓAπt)) (pr2 ΓAπt))). }
   apply weqonsecfibers; intro A.
   unfold cwf_fiber_representation.
   (* reassociation:

--- a/TypeTheory/Initiality/Typing.v
+++ b/TypeTheory/Initiality/Typing.v
@@ -60,7 +60,7 @@ Section Judgements.
 Other judgements, such as well-formed contexts and context maps, will be “auxiliary” judgements defined in terms of these afterwards.
 
 The inductive definition of derivability involves only the primitive judgements. *)
-  Inductive judgement
+  Polymorphic Inductive judgement
   :=
     | ty_judgement (Γ : context) (A : ty_expr Γ)
     | tyeq_judgement (Γ : context) (A A' : ty_expr Γ)

